src/Data/Text.hs

```diff
@@ -1070,6 +1071,11 @@ foldl1' :: HasCallStack => (Char -> Char -> Char) -> Text -> Char
 foldl1' f t = S.foldl1' f (stream t)
 {-# INLINE foldl1' #-}
 
+-- | /O(n)/ A monadic version of 'foldl'.
+foldlM :: Monad m => (a -> Char -> m a) -> a -> Text -> m a
+foldlM f z t = S.foldlM f z (stream t)
```

Comment:
Does this implementation match [`Data.Foldable.foldlM`](https://hackage.haskell.org/package/base-4.19.0.0/docs/Data-Foldable.html#v:foldlM), especially with regards to short-circuiting and the order of effects? Would be nice to have a test, at least for `m ~ Maybe` and `m ~ Either`.